include/vara/Feature/FeatureModel.h

```diff
@@ -0,0 +1,222 @@
+#ifndef VARA_FEATUREMODEL_FEATUREMODEL_H
+#define VARA_FEATUREMODEL_FEATUREMODEL_H
+
+#include "llvm/ADT/SmallSet.h"
+#include "llvm/Support/GraphWriter.h"
+
+#include "vara/Feature/Feature.h"
+
+using std::string;
+namespace vara {
+
+//===----------------------------------------------------------------------===//
+//                               FeatureModel
+//===----------------------------------------------------------------------===//
+
+class FeatureModel {
+public:
+  using FeatureMapTy = llvm::StringMap<std::unique_ptr<FMFeature>>;
+
+  using ConstraintsTy = std::vector<std::vector<std::pair<FMFeature *, bool>>>;
+
+private:
+  string Name;
+
+  FeatureMapTy Features;
+
+  ConstraintsTy Constraints;
+
+  FMFeature *Root;
+
+public:
+  FeatureModel(string VM, FeatureMapTy &Features, ConstraintsTy &Constraints)
+      : Name(std::move(VM)), Features(std::move(Features)),
+        Constraints(std::move(Constraints)), Root(Features["root"].get()) {}
```

Comment:
```suggestion
  FeatureModel(string VM, FeatureMapTy Features, ConstraintsTy Constraints)
      : Name(std::move(VM)), Features(std::move(Features)),
        Constraints(std::move(Constraints)), Root(Features["root"].get()) {}
```

If we move the objects into our own class we need to take the parameters by value